tests/cli/backup/test_backup_command.py

```diff
@@ -1,10 +1,150 @@
+    cg_context: CGConfig,
+    caplog,
+    mocker,
+    flow_cell_name: str,
+):
+    """Test encrypt flow cell in dry run mode when sequencing is not done."""
+    caplog.set_level(logging.DEBUG)
+
+    # GIVEN flow cells that are being sequenced
+    mocker.patch.object(FlowCellDirectoryData, "is_flow_cell_ready")
+    FlowCellDirectoryData.is_flow_cell_ready.return_value = False
+
+    # GIVEN a flow cells directory
+
+    # WHEN encrypting flow cells in dry run mode
+    result = cli_runner.invoke(encrypt_flow_cells, ["--dry-run"], obj=cg_context)
+
+    # THEN exits without any errors
+    assert result.exit_code == EXIT_SUCCESS
+
+    # THEN communicate flow cell is not ready
+    assert f"Flow cell: {flow_cell_name} is not ready" in caplog.text
+
+
+def test_encrypt_flow_cell_when_encryption_already_started(
+    cli_runner: CliRunner,
+    cg_context: CGConfig,
+    caplog,
+    encryption_dir: Path,
+    flow_cell_name: str,
+    flow_cell_full_name: str,
+    mocker,
+):
+    """Test encrypt flow cell in dry run mode when pending file exists"""
+    caplog.set_level(logging.DEBUG)
+
+    # GIVEN flow cells that are ready
+    mocker.patch.object(FlowCellDirectoryData, "is_flow_cell_ready")
+    FlowCellDirectoryData.is_flow_cell_ready.return_value = True
+
+    # Given a pending flag file
+    flow_cells_dir = Path(cg_context.backup.encrypt_dir, flow_cell_full_name)
+    flow_cells_dir.mkdir(parents=True, exist_ok=True)
+    Path(flow_cells_dir, flow_cell_name).with_suffix(FileExtensions.PENDING).touch()
+
+    # GIVEN a flow cells directory
+
+    # WHEN encrypting flow cells in dry run mode
+    result = cli_runner.invoke(encrypt_flow_cells, ["--dry-run"], obj=cg_context)
+
+    # THEN exits without any errors
+    assert result.exit_code == EXIT_SUCCESS
+
+    # THEN communicate flow cell encryption already started
+    assert f"Encryption already started for flow cell: {flow_cell_name}" in caplog.text
+
+
+def test_encrypt_flow_cell_when_encryption_already_completed(
+    cli_runner: CliRunner,
+    cg_context: CGConfig,
+    caplog,
+    encryption_dir: Path,
+    flow_cell_name: str,
+    flow_cell_full_name: str,
+    mocker,
+):
+    """Test encrypt flow cell in dry run mode when completed file exists"""
+    caplog.set_level(logging.DEBUG)
+
+    # GIVEN flow cells that are ready
+    mocker.patch.object(FlowCellDirectoryData, "is_flow_cell_ready")
+    FlowCellDirectoryData.is_flow_cell_ready.return_value = True
+
+    # Given a pending flag file
+    flow_cells_dir = Path(cg_context.backup.encrypt_dir, flow_cell_full_name)
+    flow_cells_dir.mkdir(parents=True, exist_ok=True)
+    Path(flow_cells_dir, flow_cell_name).with_suffix(FileExtensions.COMPLETE).touch()
+
+    # GIVEN a flow cells directory
+
+    # WHEN encrypting flow cells in dry run mode
+    result = cli_runner.invoke(encrypt_flow_cells, ["--dry-run"], obj=cg_context)
+
+    # THEN exits without any errors
+    assert result.exit_code == EXIT_SUCCESS
```

Comment:
I'm wondering if these commands that run a process over several flow cells or directories should exit successfully or take into account flow cells where the process could not be completed and abort after iterating over all flow cells. Similar to what @Vince-janv has been doing recently

Reply:
Fixed